docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,249 @@
+    Name      Status    Message              Error
+
+    etcd-0    True      {"health":"true"}
+
+
+   $ kubectl get cstorpools -n openebs # Status should be Healthy.
+
+   NAME                                ALLOCATED   FREE   CAPACITY   STATUS    READONLY   TYPE      AGE
+   cstor-pool-openebs-replica-1-w3r7   6.98G       437G   444G       Healthy   false      striped   4d22h
+
+
+   $ kubectl get cstorvolume -n openebs # Status should be Healthy.
+
+   NAME                                       STATUS    AGE     CAPACITY
+   pvc-183dc3df-a7d9-4273-a8e2-7f66d2e19f4e   Healthy   3d20h   50Gi
+   pvc-5d4b4c2b-2ed5-4e75-aeb9-fbd2918ebb78   Healthy   3d20h   50Gi
+   ```
+## Steps
+
+Set the following environment variables in the terminal to assist in the upgrade process:
+```bash
+export OPENEBS_OLD_VERSION=1.12.0
+export OPENEBS_NEW_VERSION=2.2.0
+```
+### Step 1: Upgrade OpenEBS control plane components
+
+Lokomotive provides an easy way of upgrading the OpenEBS control plane.
+
+Execute the following command to upgrade OpenEBS control plane:
+
+```bash
+lokoctl component apply openebs-operator
+```
+
+Doing so, terminates the OpenEBS resources associated with the old version and new
+resources are created for the new version.
+
+Verify all the pods are in `Running` state before proceeding:
+
+```bash
+$ kubectl get pods -n openebs -l openebs.io/version=${OPENEBS_NEW_VERSION}
+```
+
+### Step 2: Upgrade OpenEBS data plane components
+
+OpenEBS control plane and data plane components work independently. Even if the
+control plane components are upgraded to the new version, the data plane components
+continue to work with the older version.
+
+#### Upgrade cStor pools
+
+Create a Job to upgrade the existing cStor pools:
+
+```bash
+cat > upgrade-cstor-pools-from-${OPENEBS_OLD_VERSION}-to-${OPENEBS_NEW_VERSION}.yaml <<EOF
+# upgrade-cstor-pools-from-1.12.0-to-2.2.0.yaml
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cstor-spc-${OPENEBS_OLD_VERSION}-to-${OPENEBS_NEW_VERSION}
+  namespace: openebs
+spec:
+  backoffLimit: 4
+  template:
+    spec:
+      serviceAccountName: openebs-operator
+      containers:
+      - name:  upgrade
+        args:
+        - "cstor-spc"
+        - "--v=4"
+        - "--from-version=${OPENEBS_OLD_VERSION}"
+        - "--to-version=${OPENEBS_NEW_VERSION}"
+$(kubectl get spc -n openebs --no-headers -o custom-columns=":metadata.name" | sed 's/.*/        - "&"/')
+        #DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+        image: openebs/m-upgrade:${OPENEBS_NEW_VERSION}
+        imagePullPolicy: Always
+      restartPolicy: OnFailure
+EOF
+```
+
+Apply the job to start the upgrade for cStor pools:
+
+```bash
+kubectl apply -f upgrade-cstor-pools-from-${OPENEBS_OLD_VERSION}-to-${OPENEBS_NEW_VERSION}.yaml
```

Comment:
If we don't use `generateName`, we can pipe directly to `kubectl apply` right?

Reply:
https://github.com/kinvolk/lokomotive/pull/1163#discussion_r525841048